### PR TITLE
alabama scraper

### PR DESCRIPTION
**Pull request by ghost**
finally got it all working — scrapes both the RFP site and the separate ITB site.

a couple of things:
-  i added both urls to the yaml; not sure if that's how you want it
-  since this scrapes both `RFPs` and `ITBs`, i named it `solicitations.coffee` instead of `rfps.coffee`

happy to make any changes or adjustments if you'd like. just lemme know.

### CODE REVIEW

File: scrapers/al/config.yml [hunk 1]
```diff
@@ -0,0 +1,4 @@
+name: Alabama
+index_url: 
+  - http://www.purchasing.alabama.gov/txt/ITBs.aspx
```
Comment (ajb) [inline scrapers/al/config.yml]: These URLs should be encased in double quotes.

**Comment by ajb** [comment]
Hey Kiran, awesome work.

I'm looking at the output, and there are quite a few records that are too sparse on details:

``` json
  {
    "type": "RFP",
    "id": "2013-12-03",
    "title": "Appraisal Services for ALDOT, Statewide.",
    "department_name": "Transportation"
  },
  {
    "type": "RFP",
    "id": "2014-53-01",
    "title": "The Agency is requesting proposals from qualified vendors to develop, implement and maintain an Advanced Fraud Detection System including a 100% web-based case management system (hereinafter, “System”).",
    "description": "The Agency is requesting proposals from qualified vendors to develop, implement and maintain an Advanced Fraud Detection System including a 100% web-based case management system (hereinafter, “System”).  The System will leverage the newest technology, fraud and abuse detection techniques, methodologies and services available in the marketplace.  Advanced detection tools that incorporate predictive analytics and neural networking will allow the users to query the data in a way that reveals patterns and relationships between people, places, events, times, and things. They also allow queries based on groups, “nearness” and other clustered or networked behavior.  These varied analytic techniques offer users choices in how to uncover connections in seemingly unrelated data.  This will also enhance the Agency’s ability to target provider and recipient investigations, reviews and audits, ensuring a higher return on the Agency’s resource investment and assisting the Agency in identifying high",
    "department_name": "Medicaid"
  },
  {
    "type": "RFP",
    "id": "2014-19-02",
    "title": "A Pension Administration System that will modernize and streamline the non-investment core business process within the ERS, TRS, and JRF systems.",
    "department_name": "Retirement Systems of Alabama"
  }
```

Can you see if you can (at the very least) add a `html_url` property to these? A description would be great, too.

Also, if you don't mind naming the file `rfps.coffee`, I'd like to do the rfps -> solicitations renaming all at once, instead of scraper-by-scraper.

**Comment by ghost** [comment]
thanks for the quick turnaround, @adamjacobbecker, and apologies in advance for the length of this comment.

fixed the yaml, renamed solicitations.coffee -> rfps.coffee and added the `html_url` property where it exists in the RFPs table. there's unfortunately no guarantee that the `html_url` will lead to a site describing the RFP, because the alabama site doesn't put that restriction on the urls that are given with each RFP. for example, one of the URLs given with one of the RFPs is [`http://www.va.alabama.gov`](http://www.va.alabama.gov), which leads to the general organization site — not an RFP description page. (that uncertainty in where you land was my initial rationale behind not including the `html_url`.)

however, it seems like the url will in **all** cases lead **at least** to the site affiliated with the organization sponsoring the RFP, which seems like a good enough reason to have it included wherever it is available.

``` json
% cat scrapers/al/rfps.json | jq '.'                                                                                                                                    ✭
[
  {
    "html_url": "http://cpmsweb2.dot.state.al.us/ConsultMgmt/Docs/NON/05-17-2013-1.pdf",
    "department_name": "Transportation",
    "title": "Appraisal Services for ALDOT, Statewide.",
    "id": "2013-12-03",
    "type": "RFP"
  },
  {
    "html_url": "http://www.medicaid.alabama.gov/CONTENT/2.0_newsroom/2.4_Procurement.aspx",
    "department_name": "Medicaid",
    "description": "The Agency is requesting proposals from qualified vendors to develop, implement and maintain an Advanced Fraud Detection System including a 100% web-based case management system (hereinafter, “System”).  The System will leverage the newest technology, fraud and abuse detection techniques, methodologies and services available in the marketplace.  Advanced detection tools that incorporate predictive analytics and neural networking will allow the users to query the data in a way that reveals patterns and relationships between people, places, events, times, and things. They also allow queries based on groups, “nearness” and other clustered or networked behavior.  These varied analytic techniques offer users choices in how to uncover connections in seemingly unrelated data.  This will also enhance the Agency’s ability to target provider and recipient investigations, reviews and audits, ensuring a higher return on the Agency’s resource investment and assisting the Agency in identifying high",
    "title": "The Agency is requesting proposals from qualified vendors to develop, implement and maintain an Advanced Fraud Detection System including a 100% web-based case management system (hereinafter, “System”).",
    "id": "2014-53-01",
    "type": "RFP"
  },
  {
    "html_url": "http://www.cat.aum.edu/RSARFP",
    "department_name": "Retirement Systems of Alabama",
    "title": "A Pension Administration System that will modernize and streamline the non-investment core business process within the ERS, TRS, and JRF systems.",
    "id": "2014-19-02",
    "type": "RFP"
  }, ...]
```

**unfortunately, as in the screenshot below, the alabama RFP site is itself pretty sparse, so there are entries for which there is no `html_url` provided**:

image

regarding the `title` and `description` fields, this was a pretty big hack on my part. the alabama RFP site only puts one description field in the page that can be scraped (you can see this description when you mouse over the icon in the second column). other than that, there's no title or other description anywhere accessible. the `View` links point to PDFs through postBacks, so afaik there's no way for me to create a permanent link to them and the PDFs are all different formats so i can't really parse those. (if anyone knows a way to deal with these postBacks and/or PDFs, i'd be more than happy to give it a shot.)

so what i did for `title` and `description` was i took the blob in the description field for each RFP, and i always made the first sentence the `title`. if there was more than one sentence, i stuck the entire blob into the `description` field, but if there was only one sentence i didn't create a `description` field because i thought duping the two fields didn't add much value. **therefore, every JSON object in the output has a `title` property but not every JSON object has a `description` property.** very, very open to other thoughts here!

i can add the categories in the `Category` and `Sub Category` fields to each object but given that these aren't standard categories (it looks like they don't match up with NIGP codes afaict) i thought that adding them to the schema and having them be a property might just make for more cluttered faceting, since every state might have a different set of `Category` and `Sub Category` groups and you would have to dedupe all of these in order to get rid of the overlap. again, super open to other thoughts here and happy to add these fields to the schema and to this scraper if it helps!

**in any case, if there's anything else you want me to do here, please let me know — happy to make any changes or revisions!**

**Comment by ghost** [comment]
gah, rereading the above comment i'm now realizing how obnoxiously long it is. genuinely was not trying to create more work for you, adam, but i'm betting i ended up doing just that :/ 

really sorry about this. i'll aim to be much more concise from now on.

**Comment by ajb** [comment]
Don't worry about it! I feel bad for taking so long, but I promise I'll
review it soon.

**Comment by ajb** [comment]
I'm merging this in -- my take is that their RFP is supremely f'ed, so what you've done is the best we could ask for. I thought that "View" cell would be a permalink :(

Thanks again for all your effort!

**Pull request merged**